data-model/src/main/java/org/hypertrace/core/datamodel/shared/StructuredTraceGraph.java

```diff
@@ -90,21 +109,16 @@ private void buildParentChildRelationship(StructuredTrace trace) {
     }
 
     List<Entity> entities = trace.getEntityList();
-
-    for (Event event : events) {
-      eventMap.put(event.getEventId(), event);
-    }
-
     List<Edge> eventEdges = trace.getEventEdgeList();
     if (eventEdges != null) {
       for (Edge edge : trace.getEventEdgeList()) {
         Integer sourceIndex = edge.getSrcIndex();
         Integer targetIndex = edge.getTgtIndex();
         Event parentEvent = events.get(sourceIndex);
         Event childEvent = events.get(targetIndex);
-        parentToChildrenEvents.putIfAbsent(parentEvent, new ArrayList<>());
-        parentToChildrenEvents.get(parentEvent).add(childEvent);
-        childToParentEvents.put(childEvent, parentEvent);
+        parentToChildrenEvents.computeIfAbsent(parentEvent.getEventId(), k -> new ArrayList<>())
```

Comment:
I think we need to add tests for this file.
Curious question: I think, as our entity_id are based on identifying attributes, can two different entity of different types, can have the same id? If, so, can this be a problem?

Reply:
Added tests.

 > Curious question: I think, as our entity_id are based on identifying attributes, can two different entity of different types, can have the same id? If, so, can this be a problem?

Wouldn't different entity types have different identifying attributes?

Also, I think ideally(by definition of id), two different entities shouldn't be having the same id.

Reply:
>Wouldn't different entity types have different identifying attributes?
Checked for `BACKEND` and `SERVICE` type for which I had some confusion. I think, should be fine.
cc: @buchi-busireddy